docs/painless/painless-lang-spec.asciidoc

```diff
@@ -1,73 +1,46 @@
-[[painless-specification]]
+[[painless-lang-spec]]
 == Painless Language Specification
 
-Painless uses a Java-style syntax that is similar to Groovy. In fact, most
-Painless scripts are also valid Groovy, and simple Groovy scripts are typically
-valid Painless. This specification assumes you have at least a passing
-familiarity with Java and related languages.
-
-Painless is essentially a subset of Java with some additional scripting
-language features that make scripts easier to write. However, there are some
-important differences, particularly with the casting model. For more detailed
+Painless is a scripting language designed for security and performance.
+Painless syntax is similar to Java syntax along with some additional
+features such as dynamic typing, Map and List accessor shortcuts, and array
+initializers. As a direct comparison to Java, there are some important
+differences, especially related to the casting model. For more detailed
 conceptual information about the basic constructs that Java and Painless share,
 refer to the corresponding topics in the
 https://docs.oracle.com/javase/specs/jls/se8/html/index.html[Java Language
 Specification].
 
 Painless scripts are parsed and compiled using the http://www.antlr.org/[ANTLR4]
-and http://asm.ow2.org/[ASM] libraries. Painless scripts are compiled directly
-into Java byte code and executed against a standard Java Virtual Machine. This
-specification uses ANTLR4 grammar notation to describe the allowed syntax.
+and http://asm.ow2.org/[ASM] libraries. Scripts are compiled directly
+into Java Virtual Machine (JVM) byte code and executed against a standard JVM.
+This specification uses ANTLR4 grammar notation to describe the allowed syntax.
 However, the actual Painless grammar is more compact than what is shown here.
 
-[float]
-[[comments]]
-==== Comments
-
-Painless supports both single-line and multi-line comments. You can include
-comments anywhere within a script.
-
-Single-line comments are preceded by two slashes: `// comment`. They can be
-placed anywhere on a line. All characters from the two slashes to the end of
-the line are ignored.
-
-Multi-line comments are preceded by a slash-star `/*` and closed by
-star-slash `*/`. Multi-line comments can start anywhere on a line. All
-characters from the opening `/*` to the closing `*/` are ignored.
-
-*Examples:*
-
-[source,Java]
-----
-// single-line comment
-
-<code> // single-line comment
+This specification is divided into the following sections:
+
```

Comment:
The intro text isn't really necessary.

Reply:
Removed this and with the flatten structure removed the links as well as they show up in the sidebar.